tests/Controller/ProductApiTest.php

```diff
@@ -0,0 +1,249 @@
+
+        $this->client->request('GET', '/api/v1/products/-1', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_indexing_products()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $this->loadFixturesFromFile('resources/products.yml');
+
+        $this->client->request('GET', '/api/v1/products/', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'product/index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_showing_product()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $products = $this->loadFixturesFromFile('resources/products.yml');
+
+        $this->client->request('GET', '/api/v1/products/'.$products['product1']->getId(), [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'product/show_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_delete_product_if_it_does_not_exist()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        $this->client->request('DELETE', '/api/v1/products/-1', [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_delete_product()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $products = $this->loadFixturesFromFile('resources/products.yml');
+
+        $this->client->request('DELETE', '/api/v1/products/'.$products['product1']->getId(), [], [], static::$authorizedHeaderWithContentType, []);
+
+        $response = $this->client->getResponse();
+        $this->assertResponseCode($response, Response::HTTP_NO_CONTENT);
+
+        $this->client->request('GET', '/api/v1/products/'.$products['product1']->getId(), [], [], static::$authorizedHeaderWithAccept);
+
+        $response = $this->client->getResponse();
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_create_product_with_multiple_translations()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $this->loadFixturesFromFile('resources/locales.yml');
+
+        $data =
+<<<EOT
+        {
+            "code": "MUG_TH",
+            "variantSelectionMethod": "match",
```

Comment:
It should be default on the model. The problem is with the form: it contains the field, so if we do not send it - it sets null value.